File: app/src/main/java/com/solarexsoft/solarexroomwordsample/WordRepository.java
package com.solarexsoft.solarexroomwordsample;

import android.app.Application;

import androidx.lifecycle.LiveData;

import java.util.List;

/**
 * Created by Solarex on 2020/4/29/5:54 PM
 * Desc:
 */
public class WordRepository {
    private WordDao mWordDao;
    private LiveData<List<Word>> mAllWords;

    WordRepository(Application application) {
        WordRoomDatabase db = WordRoomDatabase.getDatabase(application);
        mWordDao = db.wordDao();
        mAllWords = mWordDao.getAlphabetizedWords();
    }

    LiveData<List<Word>> getALlWords() {
        return mAllWords;
    }

    void insert(Word word) {
        WordRoomDatabase.databaseWriteExecutor.execute(() -> {
            mWordDao.insert(word);
        });
    }
}

File: app/src/main/java/com/solarexsoft/solarexroomwordsample/WordRepository.kt
package com.solarexsoft.solarexroomwordsample

import androidx.lifecycle.LiveData

/**
 * Created by Solarex on 2020/4/29/4:20 PM
 * Desc:
 */
class WordRepository(private val wordDao: WordDao) {
    val allWords: LiveData<List<Word>> = wordDao.getAlphabetizedWords()

    suspend fun insert(word: Word) {
        wordDao.insert(word)
    }
}